### PR TITLE
Get the node in the same rack or zone of the new added node

### PR DESCRIPTION
**Pull request by yitzhak12**
fix https://github.com/red-hat-storage/ocs-ci/issues/5447

### CODE REVIEW

**Review by ebenahar — APPROVED** [review]

**Review by prsurve — DISMISSED** [review]

File: ocs_ci/ocs/node.py [hunk 1]
```diff
@@ -1995,3 +1999,64 @@ def add_new_nodes_and_label_after_node_failure_ipi(
...
+    node_rack_or_zone = get_node_rack_or_zone(failure_domain, node_obj)
+    wnode_names = node_names_to_search or get_worker_nodes()
+    other_wnode_names = [name for name in wnode_names if name != node_obj.name]
+    other_wnodes = get_node_objs(other_wnode_names)
+
+    other_wnodes_in_same_rack_or_zone = [
+        wnode
+        for wnode in other_wnodes
+        if get_node_rack_or_zone(failure_domain, wnode) == node_rack_or_zone
+    ]
```
Comment (PrasadDesala) [inline ocs_ci/ocs/node.py:2030]: Please raise an exception if this fails to find out an another worker node in the same rack or zone.

File: ocs_ci/ocs/node.py [hunk 2]
```diff
@@ -1995,3 +1999,64 @@ def add_new_nodes_and_label_after_node_failure_ipi(
...
+    """
+    osd_node_names = get_osd_running_nodes()
+    other_wnodes_in_same_rack_or_zone = get_other_worker_nodes_in_same_rack_or_zone(
+        failure_domain, node_obj, node_names_to_search
+    )
+
+    osd_node_in_same_rack_or_zone = None
+    for wnode in other_wnodes_in_same_rack_or_zone:
+        if wnode.name in osd_node_names:
+            osd_node_in_same_rack_or_zone = wnode
```
Comment (PrasadDesala) [inline ocs_ci/ocs/node.py:2059]: Please raise an exception if this fails to find out an another worker node in the same rack or zone.

File: ocs_ci/ocs/node.py [hunk 3]
```diff
@@ -1995,3 +1999,64 @@ def add_new_nodes_and_label_after_node_failure_ipi(
...
+    wnode_names = node_names_to_search or get_worker_nodes()
+    other_wnode_names = [name for name in wnode_names if name != node_obj.name]
+    other_wnodes = get_node_objs(other_wnode_names)
+
+    other_wnodes_in_same_rack_or_zone = [
+        wnode
+        for wnode in other_wnodes
+        if get_node_rack_or_zone(failure_domain, wnode) == node_rack_or_zone
+    ]
+    return other_wnodes_in_same_rack_or_zone
```
Comment (PrasadDesala) [inline ocs_ci/ocs/node.py:2031]: Log the filtered worker node in the logs.
Logging such messages can really save a lot of time while debugging the test failure. Please log such info wherever necessary.

File: ocs_ci/ocs/node.py (hunk 1 above)
Comment (yitzhak12) [reply ocs_ci/ocs/node.py:2030]: Why raise an exception? it is possible that we will find only one worker node in the same rack or zone.

File: ocs_ci/ocs/node.py (hunk 2 above)
Comment (yitzhak12) [reply ocs_ci/ocs/node.py:2059]: same as my previous comment.

File: ocs_ci/ocs/node.py (hunk 3 above)
Comment (yitzhak12) [reply ocs_ci/ocs/node.py:2031]: Okay, that will do.

File: ocs_ci/ocs/node.py (hunk 1 above)
Comment (yitzhak12) [reply ocs_ci/ocs/node.py:2030]: Yes, sorry. I meant that it should be fine if it also returns an empty list.

File: ocs_ci/ocs/node.py (hunk 2 above)
Comment (yitzhak12) [reply ocs_ci/ocs/node.py:2059]: It should be fine if it also returns None. Maybe we need to add it in the docstring.

**Review by PrasadDesala — APPROVED** [review]

File: ocs_ci/ocs/machine.py [hunk 4]
```diff
@@ -1059,3 +1059,42 @@ def wait_for_ready_replica_count_to_reach_expected_value(
...
+
+    """
+    current_replica_count = get_replica_count(machine_set)
+    log.info(f"Current replica count = {current_replica_count}")
+    log.info(
+        f"Wait {timeout} seconds for the current replica count to reach the "
+        f"expected value {expected_value}"
+    )
+    sample = TimeoutSampler(
+        timeout=timeout,
```
Comment (prsurve) [inline ocs_ci/ocs/machine.py:1086]: Is 180 seconds sufficient in case of VMware

Comment (yitzhak12) [reply ocs_ci/ocs/machine.py:1086]: Do you suggest increasing the timeout?

Comment (yitzhak12) [reply ocs_ci/ocs/machine.py:1086]: I have increased the timeout to 360 seconds.

**Review by prsurve — APPROVED** [review]

**Review by PrasadDesala — APPROVED** [review]

**Pull request merged**